server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java

```diff
@@ -314,4 +320,47 @@ public void mailWithBigLinesShouldBeLoadedIntoMessage() throws Exception {
             .containsExactly("(Empty)", 1010L, "", ImmutableMap.of("Date", "Tue, 14 Jul 2015 12:30:42 +0000", "MIME-Version", "1.0"), ZONED_DATE);
     }
 
+    @Test
+    public void textBodyShouldBeSetIntoMessageInCaseOfHtmlBody() throws Exception {
+        ByteArrayInputStream messageContent = new ByteArrayInputStream(("CContent-Type: text/html\r\n"
+            + "Subject: message 1 subject\r\n"
+            + "\r\n"
+            + "my <b>HTML</b> message").getBytes(Charsets.UTF_8));
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(messageContent.read())
+            .internalDate(INTERNAL_DATE)
+            .content(messageContent)
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(TestMessageId.of(2))
+            .build();
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getPreview()).isEqualTo("my HTML message");
+        assertThat(testee.getTextBody()).hasValue("my HTML message");
+        assertThat(testee.getHtmlBody()).hasValue("my <b>HTML</b> message");
+    }
+
+    @Test
+    public void textBodyShouldBeEmptyInCaseOfEmptyHtmlBodyAndEmptyTextBody() throws Exception {
+        ByteArrayInputStream messageContent = new ByteArrayInputStream(("CContent-Type: text/html\r\n"
+            + "Subject: message 1 subject\r\n").getBytes(Charsets.UTF_8));
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(messageContent.read())
+            .internalDate(INTERNAL_DATE)
+            .content(messageContent)
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(TestMessageId.of(2))
+            .build();
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getPreview()).isEqualTo(MessagePreviewGenerator.NO_BODY);
```

Comment:
:+1: you are changing the behaviour by returning NO_BODY (empty string) instead of null.
When I ask you to get rid of the null, I meant to try to use Optionals instead.
For example Inbox will try to read htmlBody. If it is not null, it will render it, else it will render textBody. So here you are making it render nothing when there is a textOnly mail.